src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.TranslationAction_Actions.cs

```diff
@@ -184,7 +214,11 @@ public override GeneratorDriver TransformGeneratorDriver(GeneratorDriver generat
                 }
             }
 
-            internal sealed class ProjectAssemblyNameAction(string assemblyName) : CompilationAndGeneratorDriverTranslationAction
+            internal sealed class ProjectAssemblyNameAction(
+                ProjectState oldProjectState,
+                ProjectState newProjectState,
+                string assemblyName)
```

Comment:
Something to think about: in this case you don't really need to capture this as a separate property -- it's now just stored by the newProjectState and you could just grab the property from it too. The document actions you still need to remember which document(s) it is, but here you don't.

Reply:
good point!